Macroist-SwiftUI/Service/ApiClient/ApiClient.swift

```diff
@@ -0,0 +1,189 @@
+    
+    private static func logout() throws -> Void {
+        return try Auth.auth().signOut()
+    }
+    
+    private static func deleteMeal(_ key: DateEntryKey, _ id: UUID) async throws {
+        guard let uid = Auth.auth().currentUser?.uid else {
+            throw AppError.authenticationError
+        }
+        
+        // Get Document Id if exists
+        guard let documentId = try await Firestore.firestore()
+            .collection(Keys.ID.DB)
+            .document(uid)
+            .collection(key.value)
+            .whereField(Keys.ID.ID, isEqualTo: id.uuidString)
+            .getDocuments()
+            .documents
+            .first?
+            .documentID else {
+            // ID doesnt exist
+            throw AppError.technicalError
+        }
+        
+        // Attempt to delete
+        try await Firestore.firestore()
+            .collection(Keys.ID.DB)
+            .document(uid)
+            .collection(key.value)
+            .document(documentId)
+            .delete()
+    }
+    
+    private static func deleteTodaysMeal(_ id: UUID) async throws {
+        return try await deleteMeal(DateEntryKey(.init()), id)
+    }
+    
+    private static func getMonthMeals(_ key: DateEntryKey) async throws -> [MacroMeal] {
+        guard let uid = Auth.auth().currentUser?.uid else {
+            throw AppError.authenticationError
+        }
+        let documents = try await Firestore.firestore()
+            .collection(Keys.ID.DB)
+            .document(uid)
+            .collection(key.value)
+            .getDocuments()
+            .documents
+        
+        var meals: [MacroMeal] = .init()
+        
+        for document in documents {
+            meals.append(try document.data(as: MacroMeal.self))
+        }
+        
+        return meals
+    }
+
+    private static func getDayMeals(_ key: DateEntryKey) async throws -> [MacroMeal] {
+        let calender = Calendar(identifier: .iso8601)
+        var dayMeals = try await ApiClient.getMonthMeals(key).filter {
+            calender.isDate(key.date, equalTo: $0.timeStamp.dateValue(), toGranularity: .day)
+        }
+        dayMeals.sort {
+            $0.timeStamp.dateValue() > $1.timeStamp.dateValue()
+        }
+        return dayMeals
+    }
+    
+    private static func addMeal(_ food: MacroMeal) async throws -> Void {
+        guard let uid = Auth.auth().currentUser?.uid else {
+            throw AppError.authenticationError
+        }
+        try await Firestore.firestore()
+            .collection(Keys.ID.DB)
+            .document(uid)
+            .collection(DateEntryKey(.init()).value)
+            .addDocument(data: food.dictionary)
+    }
+    
+    private static func updateMeal(_ food: MacroMeal) async throws {
+        guard let uid = Auth.auth().currentUser?.uid else {
+            throw AppError.authenticationError
+        }
+        
+        // Get Document Id if exists
+        guard let documentId = try await Firestore.firestore()
```

Comment:
these firestore functions are overloaded with a throws version and a non-throws version and it confuses Xcode and doesn't know which one I want since the signatures are the same. really annoying and need to dig into it